### ISSUE TITLE
JSON-RPC id field must accept strings

### ISSUE THREAD

**Issue by mattias-p** [dotse/zonemaster-backend#431]
JSON-RPC specifies that the id field must accept all three of **strings**, **numbers** and **null**. http://www.jsonrpc.org/specification#request_object

Today only **integers** are accepted.
```
sandoche@vps323914:/usr/local/share/perl/5.20.2/auto/share/dist/Zonemaster-Backend$ curl -s -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"version_info","id":"1"}' http://localhost:5000/ && echo
{"error":{"data":"/id: Expected integer - got string.\n","message":"The JSON sent is not a valid request object.","code":"-32600"},"jsonrpc":"2.0","id":null}
```

**Comment by pamasse** [comment]
Can we accept request that don't have ID ? (Only jsonprc and method key)

> If it is not included it is assumed to be a notification

**Comment by mattias-p** [comment]
Interesting point. The notification feature doesn't make much sense to me. Does JSON::RPC implement it? Maybe we could respond to all notification requests with error -32600 (Invalid Request)?

**Comment by pamasse** [comment]
Yes we could

**Comment by matsduf** [comment]
My interpretation is that this issue has been closed by #432
